tests/unit/test_config_provider.py

```diff
@@ -667,11 +717,71 @@ def _create_os_environ_patcher(self):
 
     def test_config_store_deepcopy(self):
         config_store = ConfigValueStore()
-        config_store.set_config_provider('foo', ConstantProvider('bar'))
+        config_store.set_config_provider(
+            'constant_value', ConstantProvider('ABC')
+        )
         config_store_copy = copy.deepcopy(config_store)
-        config_store_copy.set_config_provider('fizz', ConstantProvider('buzz'))
-        assert config_store.get_config_variable('fizz') is None
-        assert config_store_copy.get_config_variable('foo') == 'bar'
+        config_store_copy.set_config_provider(
+            'constant_value_copy', ConstantProvider('123')
+        )
+        assert config_store.get_config_variable('constant_value_copy') is None
+        assert config_store_copy.get_config_variable('constant_value') == 'ABC'
+
+    @pytest.fixture
+    def config_value_store(self):
+        return self._create_config_value_store()
+
+    def test_config_store_providers_not_mutated_after_merge(
```

Comment:
It might be worth seeing if we can parameterize this by the smart config variable name to make it easier to read and potentially add to in the future.